common/metrics/otel_metrics_handler.go

```diff
@@ -147,27 +204,45 @@ func (omp *otelMetricsHandler) Stop(l log.Logger) {
 	omp.provider.Stop(l)
 }
 
-// tagsToAttributes helper to merge registred tags and additional tags converting to attribute.KeyValue struct
-func tagsToAttributes(t1 []Tag, t2 []Tag, e excludeTags) []attribute.KeyValue {
+// makeSet returns an otel attribute.Set with the given tags merged with the
+// otelMetricsHandler's tags.
+func (omp *otelMetricsHandler) makeSet(tags []Tag) attribute.Set {
+	if len(tags) == 0 {
+		return omp.set
+	}
+	attrs := make([]attribute.KeyValue, 0, omp.set.Len()+len(tags))
+	for i := omp.set.Iter(); i.Next(); {
+		attrs = append(attrs, i.Attribute())
+	}
+	for _, t := range tags {
+		attrs = append(attrs, omp.tagConverter(t))
+	}
+	return attribute.NewSet(attrs...)
+}
+
+func makeInitialSet(tags map[string]string) attribute.Set {
+	if len(tags) == 0 {
+		return *attribute.EmptySet()
+	}
 	var attrs []attribute.KeyValue
+	for k, v := range tags {
+		attrs = append(attrs, attribute.String(k, v))
+	}
+	return attribute.NewSet(attrs...)
+}
 
-	convert := func(tag Tag) attribute.KeyValue {
-		if vals, ok := e[tag.Key()]; ok {
+func makeTagConverter(excludeTags excludeTags) tagConverter {
+	if len(excludeTags) == 0 {
+		return func(tag Tag) attribute.KeyValue {
+			return attribute.String(tag.Key(), tag.Value())
+		}
+	}
```

Comment:
Not sure if this branch is worth it since the code will do the same thing anyway, and it doesn't look to be optimizing much.

Reply:
true, I didn't benchmark it. but `excludeTags == nil` is the common case, and this is called a lot, so it seems reasonable

Reply:
actually I agree, it's not worth it. I'll simplify.